packages/babel-generator/src/printer.js

```diff
@@ -58,7 +58,7 @@ export default class Printer extends Buffer {
     if (needsParens) this.push(")");
 
     // end
-    this.map.mark(node, "end");
+    if (parent) this.map.mark(parent);
```

Comment:
Any number of things. The `)` in a function call for example. Printer is itself a stack walking a tree so this call just restores the source map state when the tree node is popped off the stack.